src/ir_passes/eq.ml

```diff
@@ -78,6 +88,8 @@ let array_eq_func_body : T.typ -> Ir.exp -> Ir.exp -> Ir.exp -> Ir.exp = fun t f
 
 (* Returns the new declarations, as well as a list of further types it needs *)
 
+let is_flat_alt f = T.(is_unit (normalize f.typ))
```

Comment:
Does this generalize to all singleton types, not just unit? Like the empty record, null, and singleton variants of singleton types? Not that anyone in their right mind would be using any of those I guess (except perhaps null).

Reply:
_Very_ good point. We have a `Type.is_singleton` predicate already!